MonoGame.Framework/Input/JoystickHat.cs

```diff
@@ -48,6 +48,36 @@ public ButtonState Up
             get;
             internal set;
         }
+
+        /// <summary>
+        /// Serves as a hash function for a <see cref="T:Microsoft.Xna.Framework.Input.JoystickHat"/> object.
+        /// </summary>
+        /// <returns>A hash code for this instance that is suitable for use in hashing algorithms and data structures such as a
+        /// hash table.</returns>
+        public override int GetHashCode()
+        {
+            var hash = 0;
+
+            if (Left == ButtonState.Pressed)
+                hash |= (1 << 3);
+            if (Up == ButtonState.Pressed)
+                hash |= (1 << 2);
+            if (Right == ButtonState.Pressed)
+                hash |= (1 << 1);
+            if (Down == ButtonState.Pressed)
+                hash |= (1 << 0);
+
+            return hash;
+        }
+
+        /// <summary>
+        /// Returns a <see cref="T:System.String"/> that represents the current <see cref="T:Microsoft.Xna.Framework.Input.JoystickHat"/>.
+        /// </summary>
+        /// <returns>A <see cref="T:System.String"/> that represents the current <see cref="T:Microsoft.Xna.Framework.Input.JoystickHat"/>.</returns>
+        public override string ToString()
+        {
+            return "" + (int)Left + (int)Up + (int)Right + (int)Down;
```

Comment:
> make text move while 0 and 1 keep the text stationary between value changes

Oh... duh... I missed the `(int)`.

Anyway...  my point was there is no spacing or formatting between those values.  If that was intended then fine.

I would point out that maybe this should all be defined clearly in the docs.  Someone getting `0100` from a `ToString()` would be baffled by it and would need to read the code to figure it out.